src/content_working_group/lib.rs

```diff
@@ -2055,6 +2060,29 @@ impl<T: Trait> versioned_store_permissions::CredentialChecker<T> for Module<T> {
 }
 
 impl<T: Trait> Module<T> {
+    fn ensure_member_has_no_active_application_on_opening(
+        opening_id: CuratorOpeningId<T>,
+        member_id: T::MemberId,
+    ) -> Result<(), &'static str> {
+        for (_id, curator_application) in <CuratorApplicationById<T>>::enumerate() {
```

Comment:
This iterates all applications on all openings, which is not great, or entirely safe.

At the very least there is `CuratorOpening::curator_applications` of type `BTreeSet<CuratorApplicationId>` which allows you to iterate only applications to the given curator opening. This is still not _great_, but we will do a refactor later where membership binding is done inside subsystem modules, not in membership module, which would make the lookup direct, so this may be the best we can do for now, short of introducing a temporary reverse index.